Update Spotify app module to support recent versions of Spotify.

jcsteh opened the pull request:
### Link to issue number:
None.

### Summary of the issue:

Spotify doesn't fire focus on the correct object when it comes to the foreground (e.g. after alt+tab). I fixed this for an older version of Spotify in #5439, but this is broken in the current version of Spotify.

### Description of how this pull request fixes the issue:

Spotify fires focus on an ancestor window instead of the correct focus. However, the trick we previously used to get the correct focus no longer works. Switch to a new trick which does work.

### Testing performed:

Tested with Spotify 1.0.66.478.g1296534d:

1. Verified that moving around the document in browse mode, searching and moving through a song list in focus mode does not result in focus loss.
2. Verified that alt+tabbing out of Spotify and back in again restores focus to where it last was.
3. Verified that searching and then pressing enter on an artist in the search results does not result in focus loss.

### Known issues with pull request:
None known.

### Change log entry:
In Bug Fixes:

```
- Focus is now once again restored correctly when returning to Spotify from another application. (#7689)
```

jcsteh committed:
Update Spotify app module to support recent versions of Spotify.

1. Spotify no longer fires focus on objects that die before we can fetch them, so remove the hack which worked around this.
2. Spotify still fires focus on an ancestor window instead of the correct focus. However, the trick we previously used to get the correct focus no longer works. Switch to a new trick which does work.

LeonarddeR reviewed (approved):
This bug annoys me in a major way :)

I assume using a FocusableUnfocusableContainer here doesn't do the job?

jcsteh commented:
No:
1. FocusableUnfocusableContainer just overrides setFocus. It doesn't change
the behaviour when a focus event hits that control. This makes sense, since
that would by definition mean the container was focusable, not unfocusable.
:)
2. Even if it did bounce focus events, it searches recursively for the
first focusable node. We want the node that actually has focus. Thankfully,
Chromium can answer this question via accFocus (which NVDA exposes via
activeChild).

jcsteh committed:
Spotify: Reinstate the hack working around focus objects which die before we can fetch them.

This happens far less than it does before, but it does still happen in some cases.
For example, it happens when pressing enter on an artist in search results.

jcsteh commented:
I just reinstated the hack working around focus objects which die before we can fetch them. This happens far less than it does before, but it does still happen in some cases. For example, it happens when pressing enter on an artist in search results. I'm updating the initial comment accordingly.
